### ISSUE TITLE
Filter for related resources

### ISSUE THREAD

**Issue by Pro100proff** [crnk-project/crnk-framework#389]
I have the next path to related resource "http://localhost:8080/addresses/1/computedPayLoads". For this path, i'm trying to apply a filter "http://localhost:8080/addresses/1/computedPayLoads?filter%5BreceiveDate.nano%5D%5BLT%5D=14", but as a result i get the following : 

{
  "data" : [ {
    "id" : "1",
    "type" : "ComputedPayload",
    "attributes" : {
      "computedPayloadValue" : 123.0,
      "receiveDate" : {
        "month" : "AUGUST",
        "year" : 2018,
        "dayOfMonth" : 13,
        "hour" : 17,
        "minute" : 58,
        "monthValue" : 8,
        "nano" : 676000000,
        "second" : 14,
        "dayOfWeek" : "MONDAY",
        "dayOfYear" : 225,
        "chronology" : {
          "id" : "ISO",
          "calendarType" : "iso8601"
        }
      }
    },
    "links" : {
      "self" : "http://localhost:8080/computed_payload/1"
    }
  }, {
    "id" : "2",
    "type" : "ComputedPayload",
    "attributes" : {
      "computedPayloadValue" : 123.0,
      "receiveDate" : {
        "month" : "AUGUST",
        "year" : 2018,
        "dayOfMonth" : 13,
        "hour" : 17,
        "minute" : 58,
        "monthValue" : 8,
        "nano" : 676000000,
        "second" : 14,
        "dayOfWeek" : "MONDAY",
        "dayOfYear" : 225,
        "chronology" : {
          "id" : "ISO",
          "calendarType" : "iso8601"
        }
      }
    },
    "links" : {
      "self" : "http://localhost:8080/computed_payload/2"
    }
  } ]
}

Tell me please, what am I doing wrong? Can i filter related resources? Based on the code, Crank is trying filter only parent resource addresses by id = 1. 

Thank you in advance

**Comment by Pro100proff** [comment]
No, I use the following libraries: crnk-core, crnk-home, crnk-setup-rs, crnk-security with version 2.6.20180522184741.

**Comment by remmeier** [comment]
how is the relationship setup? manually implemented or one of the two default forwarding ones?

**Comment by Pro100proff** [comment]
Relationship setup manually. My entity implementation :

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonApiResource(type = "Address" ,resourcePath = "addresses")
public class Address {

    @JsonApiId
    private Long id;

    @JsonProperty
    private String name;

    @JsonApiRelation(idField = "computedPayLoadsIds", opposite = "computed_payload")
    protected Set<ComputedPayLoad> computedPayLoads;

    @JsonApiRelationId
    private Set<Long> computedPayLoadsIds;}

And after, I created entity using the lombok library:

Address.builder()
                .id(1L)
                .name("Gogolya")
                .computedPayLoads(computedPayLoadRepository.getComputedPayLoads())
                .computedPayLoadsIds(computedPayLoadRepository.getComputedPayLoadsIds())
                .build();

**Comment by remmeier** [comment]
and how does the relationship repository look like?

**Comment by Pro100proff** [comment]
@Builder
@Data
@AllArgsConstructor
@JsonApiResource(type = "ComputedPayload", resourcePath = "computed_payload")
public class ComputedPayLoad {

    @JsonApiId
    private Long id;

    @JsonProperty
    private Double computedPayloadValue;

    @JsonProperty
    private LocalDateTime receiveDate;
}

**Comment by remmeier** [comment]
I rather meant http://www.crnk.io/releases/stable/documentation/#_relationshiprepositoryv2, you have a custom one?

**Comment by Pro100proff** [comment]
Sorry, I misunderstood. No, I use default implementation

**Comment by remmeier** [comment]
I see, GetFromOwnerStrategy is thing you looking for. For the default one is has not been implemented so far, but should be added. Meaning that querySpec parameter in querySpec has to be honored. For hasIdField it can be passed to idQuerySpec and filtered remotely. For !hasIdField in the else clause it must be filtered with QuerySpec.apply locally in-memory. PR would be welcome should you have time.

**Comment by Pro100proff** [comment]
Thanks a lot for your help! Tell me please, when are you going to fix it? Or I'll have implemented it and send you pull request on GitHub.

**Comment by remmeier** [comment]
if you have time, a PR would be great of course. There is a OwnerFowardingRelationshipRepositoryTest that could be used verify the changes.
